api-utils/JumpCloud_API_Go_Examples/DedupeSystemUsersUnixUidGuid/dedupeSystemUsersUnixUidGuid.go

```diff
@@ -0,0 +1,216 @@
+		return
+	}
+
+	// Attach to JumpCloud
+	api = jcapi.NewJCAPI(apiKey, url)
+
+	// Fetch all users in JumpCloud
+	// Send false, because we do not want tags
+	fmt.Println("Retreiving users from JumpCloud. Depending on the number of users in your organization this may take awhile")
+	userList, err := api.GetSystemUsers(false)
+
+	if err != nil {
+		fmt.Printf("Could not read system users, err='%s'\n", err)
+		return
+	}
+	fmt.Println("Users retrieved. Finding users with duplicate unix_uid and unix_guid values")
+	dupeUid, dupeGuid := findDupeUidGuids(userList)
+	printDupeUserValues(dupeUid, "unix_uid")
+	printDupeUserValues(dupeGuid, "unix_guid")
+
+	// Print out options for cleanup
+	reader := bufio.NewReader(os.Stdin)
+	fmt.Println("Now we can help you fix your duplicate values")
+	fmt.Println("---------------------------------------------")
+	fmt.Println("Option 1: Start from the duplicate value and increment until all users have unique values")
+	fmt.Println("Option 2: Start from an admin specified value and increment until all users have unique values")
+	fmt.Println("Option 3: Print all affected user ids and do nothing")
+
+	var updateErr error
+	for unixUidValue, userList := range dupeUid {
+		updateErr = fixValuesInput(reader, "unix_uid", unixUidValue, userList)
+		if updateErr != nil {
+			fmt.Printf("Could not update users, err='%s'\n", err)
+			return
+		}
+	}
+	for unixGuidValue, userList := range dupeGuid {
+		updateErr = fixValuesInput(reader, "unix_guid", unixGuidValue, userList)
+		if updateErr != nil {
+			fmt.Printf("Could not update users, err='%s'\n", err)
+			return
+		}
+	}
+}
+
+func findDupeUidGuids(userList []jcapi.JCUser) (map[string][]jcapi.JCUser, map[string][]jcapi.JCUser) {
+	dupeUid := make(map[string][]jcapi.JCUser)
+	dupeGuid := make(map[string][]jcapi.JCUser)
+
+	for _, user := range userList {
+		dupeUid[user.Uid] = append(dupeUid[user.Uid], user)
+		dupeGuid[user.Gid] = append(dupeGuid[user.Gid], user)
+	}
+
+	dupeUid = deleteSingleUserValues(dupeUid)
+	dupeGuid = deleteSingleUserValues(dupeGuid)
+
+	return dupeUid, dupeGuid
+}
+
+func deleteSingleUserValues(userValueMap map[string][]jcapi.JCUser) map[string][]jcapi.JCUser {
+	for k, v := range userValueMap {
+		if len(v) <= 1 {
+			delete(userValueMap, k)
+		}
+	}
+	return userValueMap
+}
+
+func printDupeUserValues(userValueMap map[string][]jcapi.JCUser, field string) {
+	for k, v := range userValueMap {
+		fmt.Printf("%d users have the %s value: %s\n", len(v), field, k)
+	}
+}
+
+func incrementUnixValuesOnUsers(api jcapi.JCAPI, userList []jcapi.JCUser, field string, startingValue string, dryRun bool) error {
+	fmt.Printf("Incrementing values on field: %s for %d systemusers\n", field, len(userList))
+
+	// Update User Values, leaving the first in the list alone
+	// Field on the proto is a string, so to make things easier, we convert to int
+	// so math is easier
+	intValue, _ := strconv.Atoi(startingValue)
+	for idx := range userList {
```

Comment:
Is there a downside to `for _, user := range userList {` and operating on `user` below? You'd make the function argument `[]*jcapi.JCUser` which I think you want to do anyway.